From: Jeremy Evans <code@jeremyevans.net>
Date: Wed, 15 May 2019 20:16:03 -0700
Subject: Fix setup command if format_executable is true by default

Some operating systems that support multiple versions of
ruby set --format-executable by default.  Previously, the
setup command handled things for gem but not for bundle,
resulting in the following test failure:

TestGemCommandsSetupCommand#test_env_shebang_flag:
Errno::ENOENT: No such file or directory @ rb_sysopen - /tmp/test_rubygems_30662/install/bin/bundle26
    test/rubygems/test_gem_commands_setup_command.rb:149:in `read'
    test/rubygems/test_gem_commands_setup_command.rb:149:in `test_env_shebang_flag'

This patch fixes the issue by passing the :format_executable
option when installing the bundler gem during setup.
---
 lib/rubygems/commands/setup_command.rb | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/rubygems/commands/setup_command.rb b/lib/rubygems/commands/setup_command.rb
index f5e5236..1754630 100644
--- a/lib/rubygems/commands/setup_command.rb
+++ b/lib/rubygems/commands/setup_command.rb
@@ -429,7 +429,7 @@ By default, this RubyGems will install gem as:
     Dir.chdir("bundler") do
       built_gem = Gem::Package.build(bundler_spec)
       begin
-        installer = Gem::Installer.at(built_gem, env_shebang: options[:env_shebang], install_as_default: true, bin_dir: bin_dir, wrappers: true)
+        installer = Gem::Installer.at(built_gem, env_shebang: options[:env_shebang], format_executable: options[:format_executable], install_as_default: true, bin_dir: bin_dir, wrappers: true)
         installer.install
       ensure
         FileUtils.rm_f built_gem
